src/lib/providers/CurrentDomainProvider.tsx

```diff
@@ -28,44 +30,35 @@ export const CurrentDomainContext = React.createContext({
 	paymentToken: undefined as Maybe<string>,
 	paymentTokenInfo: {} as ConvertedTokenInfo,
 });
-
 const CurrentDomainProvider: React.FC = ({ children }) => {
 	//////////////////////////
 	// Hooks & State & Data //
 	//////////////////////////
-
 	const { chainId } = useWeb3React<Web3Provider>(); // get provider for connected wallet
 
 	// Get current domain from react-router-dom
 	const { pathname } = useLocation();
-
 	const domain = zNAFromPathname(pathname);
 
 	const zna =
 		ROOT_DOMAIN +
 		(domain.length ? (IS_DEFAULT_NETWORK ? domain : '.' + domain) : '');
 
 	const domainId = getDomainId(zna);
+
 	const znsDomain = useZnsDomain(domainId, chainId || defaultNetworkId);
 	const [domainMetadata, setDomainMetadata] = usePropsState(
 		znsDomain.domainMetadata,
 	);
+
 	const appPathname = appFromPathname(pathname);
 	const app =
 		appPathname === ROUTES.STAKING
 			? ROUTES.STAKING + ROUTES.STAKING_POOLS
 			: appPathname;
 
 	// Change document title based on current network
-	if (
-		zna.length > 0 &&
-		zna !== process.env.REACT_APP_NETWORK &&
-		domainMetadata?.title
-	) {
-		document.title = process.env.REACT_APP_TITLE + ' | ' + domainMetadata.title;
-	} else {
-		document.title = process.env.REACT_APP_TITLE + ' | NFTs';
-	}
+	useDocumentTitle(zna, app, domainMetadata?.title);
```

Comment:
Can we rename this method as the name `use` suggests it to be a hook?
May be simply calling it like `updateDocumentTitle` or `modifyDocumentTitle`